metadata/client.go

```diff
@@ -1349,6 +1351,30 @@ func (fn fetchPropertiesFn) Properties() Properties {
 	return properties
 }
 
+type isEmbeddingGetter interface {
+	GetIsEmbedding() bool
+}
+
+type fetchIsEmbeddingFn struct {
+	getter isEmbeddingGetter
+}
+
+func (fn fetchIsEmbeddingFn) IsEmbedding() bool {
+	return fn.getter.GetIsEmbedding()
+}
+
+type dimensionGetter interface {
+	GetDimension() uint32
+}
+
+type fetchDimensionFn struct {
+	getter dimensionGetter
+}
+
+func (fn fetchDimensionFn) Dimension() uint32 {
```

Comment:
god I hate that this is a uint32, just cast it to an int please. I'm willing to just have this do weird shit if someone actually has a vector of a dimension that would require the 32nd bit